### ISSUE TITLE
command language not available for nvim 0.7.2 on macOS (intel chip)

### ISSUE THREAD

**Issue by jdhao** [neovim/neovim#19127]
### Neovim version (nvim -v)

0.7.2

### Vim (not Nvim) behaves the same?

N/A

### Operating system/version

macOS 12.4 (intel chip)

### Terminal name/version

wezterm

### $TERM environment variable

xterm-256color

### Installation

binary release

### How to reproduce the issue

nvim --clean -c "language"

### Expected behavior

No error occurs.

### Actual behavior

I see the following error message:

> E319: The command is not available in this version

I suspect this is an issue with mac intel chip machine, because I asked folks with 0.7.2 on M1 chip, the `:language` command is available.

**Comment by carlocab** [comment]
Odd. If anything, I would've expected `:language` to be missing on ARM and not on Intel.

**Comment by carlocab** [comment]
The quick and dirty way to fix this is to build the `nvim` binary once for each architecture then `lipo` them up.

The nice (but not quick) way to fix this is to find out why the `:language` command isn't being enabled on the `x86_64` slice of the `nvim` binary, and then fix that. Is this `src/nvim/po/CMakeLists.txt`?

**Comment by zeertzjq** [comment]
`HAVE_WORKING_LIBINTL` not being defined causes `:language` to be disabled, which seems to depend on this piece of code in `FindLibIntl.cmake`:
```cmake
check_c_source_compiles("
#include <libintl.h>

int main(int argc, char** argv) {
  gettext(\"foo\");
  ngettext(\"foo\", \"bar\", 1);
  bindtextdomain(\"foo\", \"bar\");
  bind_textdomain_codeset(\"foo\", \"bar\");
  textdomain(\"foo\");
}" HAVE_WORKING_LIBINTL)
```

**Comment by carlocab** [comment]
Oh, right. That seems simple then. Let me test out a fix and open a PR if it works.

**Comment by carlocab** [comment]
I have a hacky fix at #19138. I thought I had a proper one, but CMake refuses to cooperate.

If I run the `HAVE_WORKING_LIBINTL` test outside of CMake with the appropriate linker flags, it succeeds. I just haven't managed to convince CMake to use said appropriate linker flags.

See https://github.com/carlocab/neovim/runs/7093828788?check_suite_focus=true#step:8:2.
